src/storage/storage.cpp

```diff
@@ -575,19 +564,16 @@ Storage::ReturnCode Storage::Run(int max_wait)
     EntryClassID *entry_class_id_ptr = shared_layout_ptr->GetBlockPtr<EntryClassID, true>(
         shared_memory_ptr, SharedDataLayout::ENTRY_CLASSID);
 
-    extractor::OriginalEdgeData current_edge_data;
-    for (unsigned i = 0; i < number_of_original_edges; ++i)
-    {
-        edges_input_stream.read((char *)&(current_edge_data), sizeof(extractor::OriginalEdgeData));
-        via_geometry_ptr[i] = current_edge_data.via_geometry;
-        name_id_ptr[i] = current_edge_data.name_id;
-        travel_mode_ptr[i] = current_edge_data.travel_mode;
-        lane_data_id_ptr[i] = current_edge_data.lane_data_id;
-        turn_instructions_ptr[i] = current_edge_data.turn_instruction;
-        entry_class_id_ptr[i] = current_edge_data.entry_classid;
-        pre_turn_bearing_ptr[i] = current_edge_data.pre_turn_bearing;
-        post_turn_bearing_ptr[i] = current_edge_data.post_turn_bearing;
-    }
+    io::readEdges(edges_input_stream,
+                  via_geometry_ptr,
+                  name_id_ptr,
+                  turn_instructions_ptr,
+                  lane_data_id_ptr,
+                  travel_mode_ptr,
+                  entry_class_id_ptr,
+                  pre_turn_bearing_ptr,
+                  post_turn_bearing_ptr,
+                  number_of_original_edges);
```

Comment:
If you accidentally mix up one of these 10 parameters it will still compile but hopefully crash hard at runtime..